framework/src/main/java/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -5523,6 +5523,15 @@ public boolean wpiShouldInferTypesForReceivers() {
     return true;
   }
 
+  /**
+   * Changes the type of {@code rhsATM} when being assigned to anything, for use by whole-program
+   * inference. The default implementation does nothing.
+   *
+   * @param rhsATM the type of the rhs of the pseudo-assignment, which is side-effected by this
+   *     method
+   */
+  public void wpiAdjustForInitializationAnnotations(AnnotatedTypeMirror rhsATM) {}
```

Comment:
The name "wpiAdjustForInitializationAnnotations" doesn't convey the generic purpose of this method. Rather, it conveys the specific purpose that you've used the method for in the Nullness Checker.

The name here should be chosen to explain when this method is called and what it might be used for in the _general_ case, not the specific case that you're trying to fix in this PR. I suggest choosing a name that will help someone reading `WholeProgramInferenceImplementation#updateAnnotationSet` to understand what this hook does.